assets/js/blocks/products/all-products/block.tsx

```diff
@@ -2,23 +2,22 @@
  * External dependencies
  */
 import { Component } from '@wordpress/element';
-import PropTypes from 'prop-types';
 import { ProductListContainer } from '@woocommerce/base-components/product-list';
 import { InnerBlockLayoutContextProvider } from '@woocommerce/shared-context';
 import { gridBlockPreview } from '@woocommerce/resource-previews';
 import { StoreNoticesContainer } from '@woocommerce/blocks-checkout';
+import { Attributes as ProductListAttributes } from 'assets/js/base/components/product-list/types';
+import { BlockAttributes } from '@wordpress/blocks';
+
+interface BlockProps {
+	attributes: BlockAttributes & ProductListAttributes;
```

Comment:
It appears that the ProductListAttributes is missing certain attribute types, such as isPreview. I believe it would be beneficial to add all the missing types to the ProductListAttributes. What are your thoughts on this?

The reason we don't see any errors in VS Code is because we are using BlockAttributes & ProductListAttributes, and the BlockAttributes type is defined as:

```
export type BlockAttributes = Record<string, any>;
```

As a result, no errors are displayed. Perhaps we can consider avoiding the usage of the BlockAttributes type(unless required) to make the types more explicit and specific. What do you think?

Reply:
Ohh you're correct, I didn't realize the BlockAttributes were so generic :/, I'll be replacing it with a local implementation in the component file